From: Ladislav Michl <ladis@linux-mips.org>
Date: Tue, 9 Apr 2024 16:19:53 +0200
Subject: [PATCH] TinyXML: define TIXML_USE_STL

We build TinyXML with TIXML_USE_STL, so define it for KODI as well.
---
 xbmc/pvr/guilib/GUIEPGGridContainer.cpp | 3 +--
 xbmc/utils/XBMCTinyXML.h                | 2 ++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/xbmc/pvr/guilib/GUIEPGGridContainer.cpp b/xbmc/pvr/guilib/GUIEPGGridContainer.cpp
index de3ef1059e4d..36121dfdcaee 100644
--- a/xbmc/pvr/guilib/GUIEPGGridContainer.cpp
+++ b/xbmc/pvr/guilib/GUIEPGGridContainer.cpp
@@ -26,6 +26,7 @@
 #include "utils/MathUtils.h"
 #include "utils/StringUtils.h"
 #include "utils/Variant.h"
+#include "utils/XBMCTinyXML.h"
 
 #include <algorithm>
 #include <memory>
@@ -33,8 +34,6 @@
 #include <string>
 #include <utility>
 
-#include <tinyxml.h>
-
 using namespace KODI;
 using namespace PVR;
 
diff --git a/xbmc/utils/XBMCTinyXML.h b/xbmc/utils/XBMCTinyXML.h
index 2f4e1888da19..2b651718e53e 100644
--- a/xbmc/utils/XBMCTinyXML.h
+++ b/xbmc/utils/XBMCTinyXML.h
@@ -19,6 +19,8 @@
 #define TYPECOUNT   TINYXML_TYPECOUNT
 #endif
 
+#define TIXML_USE_STL
+
 #include <tinyxml.h>
 #include <string>
 
